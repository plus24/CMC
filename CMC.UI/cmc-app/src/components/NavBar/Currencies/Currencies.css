div.currencies div,div.currencies svg{
    color: #fff !important;
}