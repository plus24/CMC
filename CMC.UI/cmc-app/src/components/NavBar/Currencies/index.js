import { MenuItem, Select } from "@mui/material";
import React, { useContext } from "react";
import { CurrencyContext } from "../../../contexts/CurrencyContext";
import "./Currencies.css";

function Currencies() {

  const currencyContext = useContext(CurrencyContext);
  return (
    <Select
    labelId="demo-simple-select-label"
    id="demo-simple-select"
    value={currencyContext.currency}
    label="Currency"
    onChange={currencyContext.setCurrency}
    className="currencies"
  >
    <MenuItem value='AUD'>AUD</MenuItem>
    <MenuItem value='USD'>USD</MenuItem>
    <MenuItem value='EUR'>EUR</MenuItem>
  </Select>
  );
}

export default Currencies;
