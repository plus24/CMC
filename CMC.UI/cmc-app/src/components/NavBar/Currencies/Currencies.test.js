import React from 'react';
import renderer from 'react-test-renderer';
import Currencies from './index';
import CurrencyContextProvider from '../../../contexts/CurrencyContext';

// beforeEach(() => {
// });

it('renders Currencies without crashing', () => {
  const component = renderer.create(
    <CurrencyContextProvider>
      <Currencies></Currencies>
    </CurrencyContextProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});