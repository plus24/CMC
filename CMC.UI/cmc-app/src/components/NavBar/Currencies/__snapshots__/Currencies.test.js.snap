// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render of currencies component 1`] = `
<div
  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary currencies css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
  onClick={[Function]}
>
  <div
    aria-expanded="false"
    aria-haspopup="listbox"
    aria-labelledby="demo-simple-select-label demo-simple-select"
    className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
    id="demo-simple-select"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex={0}
  >
    AUD
  </div>
  <input
    aria-hidden={true}
    className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
    onAnimationStart={[Function]}
    onChange={[Function]}
    tabIndex={-1}
    value="AUD"
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
    data-testid="ArrowDropDownIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
  <fieldset
    aria-hidden={true}
    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
  >
    <legend
      className="css-173wfxe"
    >
      <span>
        Currency
      </span>
    </legend>
  </fieldset>
</div>
`;

exports[`Test render of product component 1`] = `
<div
  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary currencies css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
  onClick={[Function]}
>
  <div
    aria-expanded="false"
    aria-haspopup="listbox"
    aria-labelledby="demo-simple-select-label demo-simple-select"
    className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
    id="demo-simple-select"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex={0}
  >
    AUD
  </div>
  <input
    aria-hidden={true}
    className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
    onAnimationStart={[Function]}
    onChange={[Function]}
    tabIndex={-1}
    value="AUD"
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
    data-testid="ArrowDropDownIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
  <fieldset
    aria-hidden={true}
    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
  >
    <legend
      className="css-173wfxe"
    >
      <span>
        Currency
      </span>
    </legend>
  </fieldset>
</div>
`;

exports[`renders Currencies without crashing 1`] = `
<div
  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary currencies css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
  onClick={[Function]}
>
  <div
    aria-expanded="false"
    aria-haspopup="listbox"
    aria-labelledby="demo-simple-select-label demo-simple-select"
    className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
    id="demo-simple-select"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex={0}
  >
    AUD
  </div>
  <input
    aria-hidden={true}
    className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
    onAnimationStart={[Function]}
    onChange={[Function]}
    tabIndex={-1}
    value="AUD"
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
    data-testid="ArrowDropDownIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
  <fieldset
    aria-hidden={true}
    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
  >
    <legend
      className="css-173wfxe"
    >
      <span>
        Currency
      </span>
    </legend>
  </fieldset>
</div>
`;
